warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java

```diff
@@ -62,42 +66,68 @@ public Object visit(ASTFunctionNode node, Object data) {
-            List<JexlNode> children = new ArrayList<>(identifiers.size());
+            List<JexlNode> extractChildren = new ArrayList<>(identifiers.size());
+            List<JexlNode> keepChildren = new ArrayList<>(identifiers.size());
 
             for (ASTIdentifier identifier : identifiers) {
                 JexlNode regexNode = buildRegexNode(identifier, functionMetadata.name(), JexlNodes.getIdentifierOrLiteralAsString(arguments.get(1)));
                 if (regexNode != null) {
-                    children.add(regexNode);
+                    extractChildren.add(regexNode);
+                } else {
+                    keepChildren.add(JexlNodeFactory.buildIdentifier(identifier.getName()));
                 }
+
             }
 
-            // only re-parent if the same number of regex nodes were built
-            if (identifiers.size() == children.size()) {
-                switch (identifiers.size()) {
-                    case 0:
-                        return returnNode;
-                    case 1:
-                        if (log.isTraceEnabled()) {
-                            log.trace("Rewrote \"" + JexlStringBuildingVisitor.buildQueryWithoutParse(node) + "\" into \""
-                                            + JexlStringBuildingVisitor.buildQueryWithoutParse(children.get(0)) + "\"");
-                        }
-                        return children.get(0);
-                    default:
-                        if (functionMetadata.name().equals(INCLUDE_REGEX)) {
-                            returnNode = JexlNodeFactory.createOrNode(children);
-                        } else {
-                            // build an AND node because of how DeMorgan's law works with expanding negations
-                            returnNode = JexlNodeFactory.createAndNode(children);
-                        }
-
-                        if (log.isTraceEnabled()) {
-                            log.trace("Rewrote \"" + JexlStringBuildingVisitor.buildQueryWithoutParse(node) + "\" into \""
-                                            + JexlStringBuildingVisitor.buildQueryWithoutParse(returnNode) + "\"");
-                        }
-                        return returnNode;
+            if (extractChildren.size() == 0) {
+                // nothing to rewrite
+                return returnNode;
+            } else if (keepChildren.size() == 0) {
+                // rewrite all nodes
+                if (identifiers.size() == 1) {
+                    // we've already rewritten our one node
+                    returnNode = extractChildren.get(0);
+                } else {
+                    if (functionMetadata.name().equals(INCLUDE_REGEX) && arguments.get(0) instanceof ASTOrNode) {
+                        returnNode = JexlNodeFactory.createOrNode(extractChildren);
+                    } else {
+                        // build an AND node because of how DeMorgan's law works with expanding negations
```

Comment:
The code looks good.  However I would like a comment here making it clear that we are ignoring the and/or for the exclude regex case because of ....